modules/deployment-scripts/build-acr/main.bicep

```diff
@@ -0,0 +1,140 @@
+param existingManagedIdentityResourceGroupName string = resourceGroup().name
+
+@description('A delay before the script import operation starts. Primarily to allow Azure AAD Role Assignments to propagate')
+param initialScriptDelay string = '30s'
+
+@allowed([
+  'OnSuccess'
+  'OnExpiration'
+  'Always'
+])
+@description('When the script resource is cleaned up')
+param cleanupPreference string = 'OnSuccess'
+
+@description('The Git Repository URL, eg. https://github.com/YOURORG/YOURREPO.git')
+param gitRepositoryUrl string
+
+@description('The name of the repository branch to use')
+param gitBranch string = 'main'
+
+@description('The directory in the repo that contains the dockerfile')
+param gitRepoDirectory string = ''
+
+@description('The image name/path you want to create in ACR')
+param imageName string
+
+@description('The image tag you want to create')
+param imageTag string = string(dateTimeToEpoch(utcNow()))
+
+@description('The ACR compute platform needed to build the image')
+param acrBuildPlatform string = 'linux'
+
+var repo = '${gitRepositoryUrl}#${gitBranch}:${gitRepoDirectory}'
+var cleanRepoName = last(split(gitRepositoryUrl, '/'))
+var tag = '${imageName}:${imageTag}'
+
+resource acr 'Microsoft.ContainerRegistry/registries@2021-12-01-preview' existing = {
+  name: AcrName
+}
+
+resource newDepScriptId 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = if (!useExistingManagedIdentity) {
+  name: managedIdentityName
+  location: location
+}
+
+resource existingDepScriptId 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' existing = if (useExistingManagedIdentity ) {
+  name: managedIdentityName
+  scope: resourceGroup(existingManagedIdentitySubId, existingManagedIdentityResourceGroupName)
+}
+
+resource rbac 'Microsoft.Authorization/roleAssignments@2022-04-01' = if (!empty(rbacRoleNeeded)) {
+  name: guid(acr.id, rbacRoleNeeded, useExistingManagedIdentity ? existingDepScriptId.id : newDepScriptId.id)
+  scope: acr
+  properties: {
+    roleDefinitionId: resourceId('Microsoft.Authorization/roleDefinitions', rbacRoleNeeded)
+    principalId: useExistingManagedIdentity ? existingDepScriptId.properties.principalId : newDepScriptId.properties.principalId
+    principalType: 'ServicePrincipal'
+  }
+}
+
+resource createImportImage 'Microsoft.Resources/deploymentScripts@2020-10-01' = {
+  name: 'ACR-Build-${cleanRepoName}'
+  location: location
+  identity: {
+    type: 'UserAssigned'
+    userAssignedIdentities: {
+      '${useExistingManagedIdentity ? existingDepScriptId.id : newDepScriptId.id}': {}
+    }
+  }
+  kind: 'AzureCLI'
+  dependsOn: [
+    rbac
+  ]
+  properties: {
+    forceUpdateTag: forceUpdateTag
+    azCliVersion: '2.30.0'
+    timeout: 'PT45M'
+    retentionInterval: 'P1D'
+    environmentVariables: [
+      {
+        name: 'acrName'
+        value: acr.name
+      }
+      {
+        name: 'acrResourceGroup'
+        value: resourceGroup().name
```

Comment:
Assuming the ACR is in the same resource group could cause issues if somewhere trying to use this in a larger deployment script where the ACR is pre-existing and not in the resource group.

Reply:
I call it out in the location parameter metadata. The logical place to run this will be in the same RG.

Reply:
The more i mull over this, yes - you could be right. Working with registries in other subscriptions could also be a valid improvement for this script.

Good call!